nodes/explorer/src/api_backend/store.rs

```diff
@@ -44,3 +47,163 @@ where
         .collect();
     make_request_and_return_response!(futures::future::try_join_all(results))
 }
+
+type Depth = usize;
+
+#[derive(Deserialize)]
+pub(crate) struct BlocksQueryParams {
+    from: BlockId,
+    #[serde(default = "default_to")]
+    to: Either<BlockId, Depth>,
+}
+
+fn default_to() -> Either<BlockId, Depth> {
+    Either::Right(500)
+}
+
+pub(crate) async fn blocks<Tx, S>(
+    State(store): State<OverwatchHandle>,
+    Query(query): Query<BlocksQueryParams>,
+) -> Response
+where
+    Tx: Transaction
+        + Clone
+        + Debug
+        + Eq
+        + Hash
+        + Serialize
+        + DeserializeOwned
+        + Send
+        + Sync
+        + 'static,
+    <Tx as Transaction>::Hash: std::cmp::Ord + Debug + Send + Sync + 'static,
+    S: StorageSerde + Send + Sync + 'static,
+{
+    let BlocksQueryParams { from, to } = query;
+    // get the from block
+    let from = match storage::block_req::<S, Tx>(&store, from).await {
+        Ok(from) => match from {
+            Some(from) => from,
+            None => {
+                return IntoResponse::into_response((StatusCode::NOT_FOUND, "from block not found"))
+            }
+        },
+        Err(e) => {
+            return IntoResponse::into_response((StatusCode::INTERNAL_SERVER_ERROR, e.to_string()))
+        }
+    };
+
+    // check if to is valid
+    match to {
+        Either::Left(to) => match storage::block_req::<S, Tx>(&store, to).await {
+            Ok(to) => match to {
+                Some(to) => handle_to::<S, Tx>(store, from, to).await,
+                None => IntoResponse::into_response((StatusCode::NOT_FOUND, "to block not found")),
+            },
+            Err(e) => {
+                IntoResponse::into_response((StatusCode::INTERNAL_SERVER_ERROR, e.to_string()))
+            }
+        },
+        Either::Right(depth) => handle_depth::<S, Tx>(store, from, depth).await,
+    }
+}
+
+async fn handle_depth<S, Tx>(
+    store: OverwatchHandle,
+    from: Block<Tx, Certificate>,
+    depth: usize,
+) -> Response
+where
+    Tx: Transaction
+        + Clone
+        + Debug
+        + Eq
+        + Hash
+        + Serialize
+        + DeserializeOwned
+        + Send
+        + Sync
+        + 'static,
+    <Tx as Transaction>::Hash: std::cmp::Ord + Debug + Send + Sync + 'static,
+    S: StorageSerde + Send + Sync + 'static,
+{
+    let mut cur = Some(from.header().parent());
```

Comment:
Nitpick: is this meant to be `current` or `cursor`? I would rather have a clear naming on things